### PR TITLE
Issue 4 create entry

### PR DESCRIPTION
**Pull request by josh-tjiauw**
https://i.imgur.com/T6xFNuR.gif

### CODE REVIEW

File: js/data.js [hunk 1]
```diff
@@ -12,6 +12,12 @@ var data = {
   entries: []
 };
 
+var newEntry = {
+  imgUrl: '',
+  entryTitle: '',
+  entryNotes: ''
+};
+
```
Comment (shadorki) [inline js/data.js]: This is not needed here.

File: js/data.js [hunk 2]
```diff
@@ -22,3 +28,14 @@ window.addEventListener('beforeunload', function(event){
...
+var previousNewEntryJSON = localStorage.getItem('newEntry');
+if (previousNewEntryJSON !== null) {
+  newEntry = JSON.parse(previousNewEntryJSON);
+}
+
+window.addEventListener('beforeunload', function (event) {
+  event.preventDefault();
+  var newEntryJSON = JSON.stringify(newEntry);
+  localStorage.setItem('newEntry', newEntryJSON);
+})
```
Comment (shadorki) [inline js/data.js]: Your `data.js` file should only contain your data model, you can move that rest of the code carefully

File: js/main.js [hunk 3]
```diff
@@ -4,16 +4,31 @@ var $username = document.getElementById('username');
...
+var $entries = document.getElementById('entries');
+var $createEntry = document.getElementById('create-entry');
+var $imgUrl = document.getElementById('imgUrl');
+var $entryPicture = document.getElementById('entryPicture');
+var $entryForm = document.getElementById('entryForm');
+var $entryTitle = document.getElementById('entryTitle');
+var $entryNotes = document.getElementById('entryNotes');
+var phImage = 'images/placeholder-image-square.jpg'
+var allImages = document.querySelectorAll('img');
+allImages.onerror = function(){this.src=phImage}
```
Comment (shadorki) [inline js/main.js]: indentation in your function.

File: js/main.js [hunk 4]
```diff
@@ -153,3 +168,31 @@ $profileButton.setAttribute('data-view', 'profile');
...
+
+$imgUrl.addEventListener('input', function(event){
+  $entryPicture.src = event.target.value;
+})
+
+$entryForm.addEventListener('submit', function (event) {
+  event.preventDefault();
+  newEntry['imgUrl'] = $imgUrl.value;
+  newEntry['entryTitle'] = $entryTitle.value;
+  newEntry['entryNotes'] = $entryNotes.value;
```
Comment (shadorki) [inline js/main.js]: You seem to be storing the newEntry object in the global scope and then overwriting it when there is a new entry.  It is better practice to create an object and store it in your data model.  If you choose not to fix this now, you will in the next feature.

```
var entry = {
  imgUrl: $imgUrl.value,
  title: $entryTitle.value,
  notes: $entryNotes.value
}
data.entries.push(entry)

**Review by shadorki — CHANGES_REQUESTED** [review]

File: js/data.js (hunk 1 above)
Comment (josh-tjiauw) [reply js/data.js]: Deleted!

File: js/main.js (hunk 4 above)
Comment (josh-tjiauw) [reply js/main.js]: image

File: js/data.js (hunk 2 above)
Comment (josh-tjiauw) [reply js/data.js]: image

File: js/main.js (hunk 3 above)
Comment (josh-tjiauw) [reply js/main.js]: image

**Review by shadorki — APPROVED** [review]

**Pull request merged**